tests/base/src/test/java/org/wildfly/security/http/client/ElytronHttpClientTest.java

```diff
@@ -0,0 +1,124 @@
+import org.wildfly.security.http.digest.DigestMechanismFactory;
+import org.wildfly.security.http.impl.AbstractBaseHttpTest;
+import org.wildfly.security.password.WildFlyElytronPasswordProvider;
+
+import java.net.URL;
+import java.net.http.HttpRequest;
+import java.security.PrivilegedAction;
+import java.security.Provider;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Supplier;
+
+import static java.security.AccessController.doPrivileged;
+
+/**
+ * Test for the ElytronHttpClient class
+ *
+ * @author <a href="mailto:kekumar@redhat.com">Keshav Kumar</a>
+ */
+public class ElytronHttpClientTest extends AbstractBaseHttpTest {
+
+    private static final String NAME = ElytronHttpClientTest.class.getSimpleName();
+
+    public static Supplier<Provider[]> ELYTRON_PASSWORD_PROVIDERS = () -> new Provider[]{
+            WildFlyElytronPasswordProvider.getInstance()
+    };
+    protected HttpServerAuthenticationMechanismFactory basicFactory = new BasicMechanismFactory(ELYTRON_PASSWORD_PROVIDERS.get());
+    protected HttpServerAuthenticationMechanismFactory digestFactory = new DigestMechanismFactory(ELYTRON_PASSWORD_PROVIDERS.get());
+    ElytronHttpClient elytronHttpClient = new ElytronHttpClient();
+
+    @Test
+    public void testRequest() throws Exception{
+        AuthenticationContextConfigurationClient AUTH_CONTEXT_CLIENT =
+                doPrivileged((PrivilegedAction<AuthenticationContextConfigurationClient>) AuthenticationContextConfigurationClient::new);
+
+        AuthenticationContext context = doPrivileged((PrivilegedAction<AuthenticationContext>) () -> {
+            try {
+                URL config = getClass().getResource("wildfly-config-http-client-basic.xml");
+                return ElytronXmlParser.parseAuthenticationClientConfiguration(config.toURI()).create();
+            } catch (Throwable t) {
+                throw new InvalidAuthenticationConfigurationException(t);
+            }
+        });
+        context.run(() -> {
+            try {
+                HttpServerAuthenticationMechanism mechanism = basicFactory.createAuthenticationMechanism("BASIC", Collections.emptyMap(),getCallbackHandler("user1", "test-realm", "password1"));
+                HttpRequest request = elytronHttpClient.getRequest("http://localhost:8080/servlet-security/SecuredServlet");
+
+                //Test successful authentication
+                TestingHttpServerRequest testingHttpServerRequest = new TestingHttpServerRequest(new String[]{request.headers().allValues("Authorization").get(0)});
+                mechanism.evaluateRequest(testingHttpServerRequest);
+                Assert.assertEquals(Status.COMPLETE,testingHttpServerRequest.getResult());
+            }catch (Exception e){
+                throw new RuntimeException(e);
+            }
+        });
+    }
+
+    @Test
+    public void testRequest2() throws Exception{
```

Comment:
Just a total minor, we can rename these tests so the name expresses what is being tested